experimental/GModule/GModule.jl

```diff
@@ -20,7 +21,18 @@ function GAP.gap_to_julia(::Type{QabElem}, a::GAP.GapObj) #which should be a Cyc
   return z
 end
 
+function GAP.gap_to_julia(::Type{QabElem}, a::Integer)
+  E = abelian_closure(QQ)[1]
+  return E(fmpz(a))
+end
+
```

Comment:
Usually we also add a type constructor for convenience:
```suggestion

QabElem(a::GapObj) = GAP.gap_to_julia(QabElem, a)

```
BTW, you could just move all these conversions into `src/GAP/gap_to_oscar.jl` right away, at least I see no reason why not to do that?

Reply:
There is already a method `QabElem(cyc::GapInt)` in `src/Groups/group_characters.jl`, I can move it to `src/GAP/gap_to_oscar.jl`.

Reply:
No, sorry, this is again one of these cases where one cannot simply move code to other files:

`QabElem` gets defined in `src/Rings/AbelianClosure.jl`.
The code in this file belongs to the module `Oscar.AbelianClosure`.
In order to install a `QabElem` method in `src/GAP/gap_to_oscar.jl`, one has to define `QabElem` in a file that is read before `src/GAP/gap_to_oscar.jl`.
Simply trying to move the `include` for `src/Rings/AbelianClosure.jl` more up is not a sensible idea, we do not have a natural linear order of code files.
We should put the relevant type definitions into new files (here `src/Rings/types.jl`, analogous to `src/Groups/types.jl`) that are read early enough.
However, then I do not see how can we achieve that the types belong to "their" modules --or is it o.k. to give this relation up?

Reply:
(In this special case, reordering some files worked because the files in `src/GAP` do not introduce new types, thus they can be read late, see PR #853.)